librz/config/config.c

```diff
@@ -386,14 +399,76 @@ static bool __is_true_or_false(const char *s) {
 	return s && (!rz_str_casecmp(s, "true") || !rz_str_casecmp(s, "false"));
 }
 
+RZ_API RzConfigNode *rz_config_set_b(RzConfig *cfg, const char *name, bool value) {
+	RzConfigNode *node = NULL;
+	char *ov = NULL;
+	ut64 oi = 0;
+
+	rz_return_val_if_fail(cfg && cfg->ht, NULL);
+	rz_return_val_if_fail(RZ_STR_ISNOTEMPTY(name), NULL);
+
+	node = rz_config_node_get(cfg, name);
+	if (node) {
+		if (rz_config_node_is_ro(node)) {
+			RZ_LOG_DEBUG("(error: '%s' config key is read only)\n", name);
+			return node;
+		}
+
+		ov = node->value;
+		oi = node->i_value;
+		if (rz_config_node_is_bool(node)) {
+			node->i_value = value ? 1 : 0;
+			char *svalue = strdup(rz_str_bool(value));
+			if (svalue) {
+				free(node->value);
+				node->value = svalue;
+			}
+		} else {
+			RZ_LOG_ERROR("(error: '%s' is not a boolean variable)\n", name);
+			return NULL;
+		}
+	} else {
+		if (!cfg->lock) {
+			node = rz_config_node_new(name, rz_str_bool(value));
+			if (!node) {
+				node = NULL;
+				goto beach;
+			}
+			node->flags = CN_RW | CN_BOOL;
+			node->i_value = value ? 1 : 0;
+			ht_pp_insert(cfg->ht, node->name, node);
+			if (cfg->nodes) {
+				rz_list_append(cfg->nodes, node);
+			}
+		} else {
+			RZ_LOG_ERROR("(locked: no new keys can be created (%s))\n", name);
+		}
+	}
+
+	if (node && node->setter) {
+		if (!node->setter(cfg->user, node)) {
+			if (oi != UT64_MAX) {
+				node->i_value = oi;
+			}
+			free(node->value);
+			node->value = strdup(ov ? ov : "");
```

Comment:
UAF reported by clang here.

Reply:
@XVilka have you seen this?

Reply:
Fixed, also fixed in another function with the same pattern.